realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/SchemaTests.kt

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import io.realm.SyncTestUtils.Companion.createTestUser
+import io.realm.entities.StringOnly
+import junit.framework.Assert
+import junit.framework.TestCase
+import org.junit.*
+import org.junit.runner.RunWith
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+class SchemaTests {
+    @get:Rule
+    val configFactory = TestSyncConfigurationFactory()
+
+    private var config: SyncConfiguration? = null
+    private var app: TestRealmApp? = null
+
+    @Before
+    fun setUp() {
+        app = TestRealmApp()
+        val user = createTestUser(app!!)
+        config = configFactory.createSyncConfigurationBuilder(user).build()
+    }
+
+    @After
+    fun tearDown() {
+        if (app != null) {
+            app!!.close()
+        }
+    }
+
+    @Test
+    fun instance() {
+        val realm = Realm.getInstance(config!!)
+        realm.use {
```

Comment:
Normally we closed any Realms in the tearDown method instead of using try-with-resources. I do realize that it is a lot easier to express in Kotlin, but unless we decide to change it everywhere I think we should stick to the same pattern in this class. Thoughts?

Reply:
The reason was just to get a feeling whether it was nicer/better/etc., and to get some insight for future API decisions. There are certainly also cases which favor explicit try-finally way of doing it, maybe even this specific case, as we need to access it after its closed. 

Otherwise I think `use` would be the right way to go for inlining `getInstance` with the block that uses it. I would not deem it necessary to change everywhere just for changing it. But I can reapply the conventional pattern if you find it better to stick to one pattern.

Reply:
True, it is being accessed after being closed in this test. I actually missed that 👍. In that case, using `realm.use { }` is probably the most appropriate here.

Generally, we have closed Realm instances in `tearDown` so you didn't have to remember it for every single test, but this class might be a little different

But is there a reason you didn't use `realm.use` here: https://github.com/realm/realm-java/pull/6850/files#diff-a8a31a8a737fec196f4a54bda8e2c9aeR61 ?

Reply:
Yes, I stopped applying it after I realized that we should maybe discuss the approach. 

I guess `use` is exactly solving the problem of not forgetting it as it is automatically closed when leaving the block. To handle it in `tearDown` we would then need to have it as a property with late init, etc. 

The motivation for `use` in this test, was explicitly that we honoured the contract of closing it if people actually uses `use`.

Reply:
Yeah, In java it was a bit easier, because you could just have a null-check in `tearDown`... Checking if the lateinit var is initialized feels a bit like a hack.

I suspect this is a case where we need to experiment a little to see what makes sense and then we probably need a roundtable discussion to settle on something.

But for now, this is fine 👍

Reply:
I ended up applying the `use` pattern throughout the file, as some of the old tests were not always calling `realm.close()` on exceptions which caused the overall execution to fail. We can then use it as input for discussions and revert if needed.